third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -824,7 +837,11 @@ impl Interpreter {
                 )
             },
         };
-        let struct_name = &*loader.name_cache.idx_to_identifier(struct_idx);
+        let struct_name_index_map = match loader {
```

Comment:
It is not that straightforward because we may also check if we allowed to read from address/module, and this information is not easy to get from interned struct identifier. We need module id interning as well, so that we can construct interned module id from interned struct identifier fast.